### PR TITLE
Support for implementing handler with an actor

### PR DESCRIPTION
**Pull request by patriknw**
* lifecycle managed by ProjectionBehavior
* the Behavior is spawned as a child of the ProjectionBehavior

This is an alternative to https://github.com/akka/akka-projection/pull/116
The difference is that the users have to implement the `ask` themselves.

Refs #144

### CODE REVIEW

File: examples/src/test/scala/docs/cassandra/WordCountDocExample.scala [hunk 1]
```diff
@@ -162,4 +168,198 @@ object WordCountDocExample {
...
+
+      private implicit val askTimeout: Timeout = 5.seconds
+
+      override def process(envelope: WordEnvelope, actor: ActorRef[WordCountProcessor.Command]): Future[Done] = {
+        actor.ask[Try[Done]](replyTo => WordCountProcessor.Handle(envelope, replyTo)).map {
+          case Success(_)   => Done
+          case Failure(exc) => throw exc
+        }
+      }
+    }
```
Comment (patriknw) [inline examples/src/test/scala/docs/cassandra/WordCountDocExample.scala:274]: This is the main API difference compared to #116
Here the users have to implement the ask themselves. Note that it requires ActorSystem in implicit scope.

File: akka-projection-core/src/main/scala/akka/projection/Projection.scala [hunk 2]
```diff
@@ -84,6 +85,11 @@ trait Projection[Envelope] {
   @InternalApi
   private[projection] def mappedSource()(implicit system: ActorSystem[_]): Source[Done, _]
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def actorHandlerInit[T]: Option[ActorHandlerInit[T]]
```
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-projection-core/src/main/scala/akka/projection/Projection.scala:91]: I created #227 for that and the other internal methods in Projection

File: akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala [hunk 3]
```diff
@@ -138,6 +139,9 @@ private[projection] class SlickProjectionImpl[Offset, Envelope, P <: JdbcProfile
   override def withStatusObserver(observer: StatusObserver[Envelope]): SlickProjectionImpl[Offset, Envelope, P] =
     copy(statusObserver = observer)
 
+  private[akka] def actorHandlerInit[T]: Option[ActorHandlerInit[T]] =
+    None // ActorHandler not supported by SlickProjection (doesn't make sense for DBIO handler)
```
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala:143]: I added a comment to https://github.com/akka/akka-projection/issues/68#issuecomment-643152448

**Comment by patriknw** [comment]
I have completed this. Ready for final review.

The java doc snippets are missing, but I'd like to merge https://github.com/akka/akka-projection/pull/233 first.

**Comment by renatocaval** [comment]
Do you want to hold this one until we merge #233? Or are you saying that you will add the snippets after this and #233 are merged?

**Comment by patriknw** [comment]
doesn’t matter which order, just didn’t want to change at both places.

File: akka-projection-core/src/main/scala/akka/projection/javadsl/ActorHandler.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+   * of the `envelope` has finished. It will not be invoked with the next envelope until after the returned
+   * `CompletionStage` has been completed.
+   *
+   * The `behavior` is spawned when the `Projection` is started and the `ActorRef` is passed in as
+   * a parameter here.
+   *
+   * You will typically use the `AskPattern.ask` to delegate the processing of the `envelope` to
+   * the actor and the returned `CompletionStage` corresponds to the reply of the `ask`.
+   */
+  def process(envelope: Envelope, actor: ActorRef[T]): CompletionStage[Done]
```
Comment (renatocaval) [inline akka-projection-core/src/main/scala/akka/projection/javadsl/ActorHandler.scala:36]: when reviewing #233, I realised that `StatefulHandler.process(State, Envelope)`

The Jdbc Session is also passed as first arg. 

This also aligns with `EventSourcedBehavior` where we have `(State, Command)` and `(State, Event)`. The most ephemeral last.

**Comment by patriknw** [comment]
ok, I have changed the order of the parameters

**Pull request merged**